chain-storage/src/error.rs

```diff
@@ -1,16 +1,18 @@
 use std::{error, fmt};
 
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug)]
 pub enum Error {
     BlockNotFound, // FIXME: add BlockId
     CannotIterate,
+    BackendError(Box<dyn std::error::Error + Send + Sync>),
```

Comment:
is it worth it to keep the error in the native format (and lose the derives), instead of flatting it into something simple ? The fact that it's handed this way just in the next chunk makes me think that just having a String of what this internal error is about, will be useful enough for debugging

Reply:
It's not typical for errors to implement `PartialEq`, precisely because of the error information baggage. If we want to be really thorough, we sould have an `ErrorKind` enum as a property, and maybe switch to `snafu` for all-bells-and-whistles error utilities.

Reply:
I agree with @mzabaluev , this is error chaining and there are lot of package/standard way to do this kind of thing (like [error-chain](https://crates.io/crates/error-chain))